File: python/easy/1470_shuffle_the_array.py
# https://leetcode.com/problems/shuffle-the-array/description/
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        newNums = [None] * 2*n
        index = 0
        for i in range (0, 2*n, 2):
            newNums[i] = (nums[index])
            index += 1
        index = 0
        for i in range (1, 2*n, 2):
            newNums[i] = nums[n + index]
            index += 1
        nums = newNums
        return nums

File: cpp/easy/1470_shuffle_the_array.cpp
// https://leetcode.com/problems/shuffle-the-array/description/
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        // TC: O(n) - SC: O(n) (in-place)
        for(int i = 0; i < n; i++){
            nums[i] += 10000 * nums[i + n];
        }
        for(int i = n - 1; i >= 0; i--){
            nums[2 * i + 1] = nums[i] / 10000;
            nums[2 * i] = nums[i] % 10000;
        }
        return nums;

        // // TC: O(n) - SC: O(2n)
        // vector<int> res;
        // for(int i = 0; i < n; i++){
        //     res.push_back(nums[i]);
        //     res.push_back(nums[i+n]);
        // }
        // return res;
    }
};
